xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -316,13 +319,14 @@ CGDisplayModeRef BestMatchForMode(
 
     if ((CGDisplayModeGetWidth(mode) == width) && (CGDisplayModeGetHeight(mode) == height))
     {
-      CGDisplayModeRelease(displayMode); // release the copy we got before ...
       displayMode = mode;
       match = true;
       break;
     }
   }
 
+  CGDisplayModeRelease(displayMode); // release the copy we got before ...
```

Comment:
releasing at this point is incorrect because you would return garbage.

this code should be carefully reviewed regarding the reference counts including the calling code.